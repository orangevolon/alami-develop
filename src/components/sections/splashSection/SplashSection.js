import { withDOM, withPubSub } from "../../../hocs";
import "./SplashSection.scss";
import avatarSrc from "../../../assets/picture.jpg";
import { PUBSUB_SCROLL } from "../../../constants";

const SplashSection = ({ document, pubSub }) => {
  const element = document.createElement("section");
  element.className = "splash-section__section";

  const pictureElement = document.createElement("img");
  pictureElement.className = "picture__img";
  pictureElement.src = avatarSrc;
  pictureElement.width = 200;
  pictureElement.height = 200;
  element.appendChild(pictureElement);

  const contentWrapper = document.createElement("div");
  contentWrapper.className = "content-wrapper__div";
  element.appendChild(contentWrapper);

  const titleElement = document.createElement("h1");
  titleElement.className = "title__h1";
  titleElement.innerText = "Amir Alami";
  contentWrapper.appendChild(titleElement);

  const age = Math.floor(
    (Date.now() - new Date(1989, 6, 15)) / (1000 * 3600 * 24 * 365)
  );

  const descriptionElement = document.createElement("p");
  descriptionElement.className = "description__p";
  descriptionElement.innerHTML = `
    Senior Full Stack Developer<br/>
    ${age} years old,<br/>
    +10 years of experience
  `;
  contentWrapper.appendChild(descriptionElement);

  pubSub.subscribe(PUBSUB_SCROLL, e => {
    pictureElement.style.transform = `translateY(${e / 2}px)`;
    pictureElement.style.filter = `blur(${e / 10}px)`;
    element.style.clipPath = `circle(${20000 / (e + 1)}px at center)`;
  });

  return element;
};

export default withPubSub(withDOM(SplashSection));
