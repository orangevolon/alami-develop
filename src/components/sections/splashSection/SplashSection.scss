@import "@assets/styles/grid.scss";
@import "@assets/styles/section.scss";

.splash-section__section {
  @include section;

  height: calc(100vh - 10rem);
  display: flex;
  overflow: hidden;

  .content-wrapper__div {
    @include content-grid;

    margin-top: auto;
    margin-bottom: 0;
  }

  .title__h1 {
    grid-area: title;
    position: relative;
    color: white;
    font-size: 3rem;
    text-transform: uppercase;
  }

  .description__p {
    grid-area: description;
    position: relative;
    color: white;
  }

  .picture__img {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    object-fit: cover;
    opacity: 0.2;
  }

  &:before {
    display: block;
    content: "";
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background: linear-gradient(90deg, #419fd9 0%, #023440 100%);
  }
}
