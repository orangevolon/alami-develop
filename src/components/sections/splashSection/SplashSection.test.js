import SplashSection from "./SplashSection";

jest.mock("../../../hocs", () => ({
  withPubSub: Component => Component,
  withDOM: Component => props =>
    Component({ ...props, document: global.document })
}));

describe("Testing components/sections/splashSection", () => {
  const pubSub = { subscribe: jest.fn() };

  it("Should render component correctly", () => {
    expect(SplashSection({ pubSub })).toMatchSnapshot();
  });

  it("Should subscribe to pubSub", () => {
    SplashSection({ pubSub });
    expect(pubSub.subscribe).toBeCalledWith(
      expect.any(String),
      expect.any(Function)
    );
  });
});
