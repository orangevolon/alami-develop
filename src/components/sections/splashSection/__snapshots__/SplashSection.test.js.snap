// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing components/sections/splashSection Should render component correctly 1`] = `
<section
  class="splash-section__section"
>
  <img
    class="picture__img"
    height="200"
    src="[object Object]"
    width="200"
  />
  <div
    class="content-wrapper__div"
  >
    <h1
      class="title__h1"
    />
    <p
      class="description__p"
    >
      
    Senior Full Stack Developer
      <br />
      
    30 years old,
      <br />
      
    +10 years of experience
  
    </p>
  </div>
</section>
`;
